src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -303,6 +307,14 @@ private boolean isReadyToBeProcessed(CloudBlockBlob blob) {
         return blob.getProperties().getLastModified().before(cutoff);
     }
 
+    private boolean isReadyToBeDeleted(Envelope envelope) {
```

Comment:
I agree with this change, it should have never been the responsibility of the enum to decide. This, however, exposes the lack of testing based on the status. Still, there's a story to address testability of the job.